yeonjae/programmers-2/주차요금계산.js

```diff
@@ -0,0 +1,60 @@
+const solution = (fees, records) => {
+  const [limitTime, feePerLimit, minute, feePerMinute] = fees;
+  const answer = [];
+  const carsNumber = [];
+  const inCars = new Map();
+  const outCars = new Map();
+
+  for (const record of records) {
+    const [time, number, state] = record.split(" ");
+
+    carsNumber.push(number);
+
+    if (state === "IN") {
+      inCars.set(number, time);
+    } else {
+      const [inHour, inMinute] = inCars.get(number).split(":");
+      const [outHour, outMinute] = time.split(":");
+      const inTotal = parseInt(inHour) * 60 + parseInt(inMinute);
+      const outTotal = parseInt(outHour) * 60 + parseInt(outMinute);
+      const savedTime = outCars.get(number) || 0;
+
+      outCars.set(number, savedTime + (outTotal - inTotal));
+      inCars.delete(number);
+    }
+  }
+
+  for (const [key, value] of inCars) {
+    const [inHour, inMinute] = value.split(":");
+    const savedTime = outCars.get(key) || 0;
+
+    outCars.set(
+      key,
+      savedTime + (60 * 23 + 59) - (parseInt(inHour) * 60 + parseInt(inMinute))
```

Comment:
```suggestion
      // 최상단 변수 선언 부
      const LAST_TIME = 60 * 23 + 59;
      // ...
      savedTime + LAST_TIME - (parseInt(inHour) * 60 + parseInt(inMinute))
```
- 60 * 23 + 59와 같은 고정적인 값은 하드코딩 하기보다 최상단에 상수로서 활용하면  더 가독성 좋은 코드가 가능합니다!

Reply:
코멘트감사합니다! 확인했습니다 :)